File: amazia-console/app/ProductImage/Controller/CreateProductImageController.php
<?php

namespace App\ProductImage\Controller;

use App\ProductImage\Service\CreateProductImageService;
use Illuminate\Http\JsonResponse;
use Illuminate\Http\Request;
use Illuminate\Routing\Controller;

class CreateProductImageController extends Controller
{
    public function __construct(private CreateProductImageService $createProductImageService) {}

    public function __invoke(Request $request, int $id): JsonResponse
    {
        $response = $this->createProductImageService->create($id, $request->file('image'));
        return response()->json($response->json(), $response->status());
    }
}

File: amazia-core/src/main/java/com/example/product/controller/CreateProductImageController.java
package com.example.product.controller;

import com.example.product.entity.ProductImage;
import com.example.product.service.CreateProductImageService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;

@RestController
@RequestMapping("/api")
public class CreateProductImageController {

    private final CreateProductImageService createProductImageService;

    public CreateProductImageController(CreateProductImageService createProductImageService) {
        this.createProductImageService = createProductImageService;
    }

    @PostMapping("/products/{id}/images")
    public ResponseEntity<ProductImage> create(
            @PathVariable Long id,
            @RequestParam("image") MultipartFile image) throws IOException {
        return ResponseEntity.status(HttpStatus.CREATED).body(createProductImageService.create(id, image));
    }
}
